src/Core/src/Handlers/Button/ButtonHandler.Android.cs

```diff
@@ -156,10 +157,23 @@ void OnNativeViewFocusChange(object? sender, AView.FocusChangeEventArgs e)
 
 		void OnPlatformViewLayoutChange(object? sender, AView.LayoutChangeEventArgs e)
 		{
-			if (sender is MaterialButton platformView && VirtualView != null)
+			UpdateIconSize();
+
+			if (sender is MaterialButton platformView && VirtualView is not null)
 				platformView.UpdateBackground(VirtualView);
 		}
 
+		void UpdateIconSize()
+		{
+			if (PlatformView is null || PlatformView.IsDisposed())
+				return;
+
+			if (PlatformView.Height > PlatformView.Width)
+				PlatformView.IconSize = PlatformView.Width - (PlatformView.PaddingLeft + PlatformView.PaddingRight);
```

Comment:
Does IconSize affect the layout at all? Is there a potential for setting the icon size trigger a new layout creating a layout loop?

Reply:
No, is not invalidating and invoking the code several times etc.